presto-tests/src/main/java/com/facebook/presto/tests/statistics/StatisticsAssertion.java

```diff
@@ -60,59 +53,82 @@ public void close()
+            estimate(nullsFraction(columnName), strategy);
+            estimate(distinctValuesCount(columnName), strategy);
+            estimate(lowValue(columnName), strategy);
+            estimate(highValue(columnName), strategy);
+            return this;
         }
 
-        public Result estimate(Metric metric, MetricComparisonStrategy strategy)
+        public <T> Checks estimate(Metric<T> metric, MetricComparisonStrategy<T> strategy)
         {
-            return testMetrics(metric, metricComparison -> metricComparison.result(strategy) == MATCH);
+            checks.add(new MetricsCheck<>(metric, strategy, MATCH));
+            return this;
         }
 
-        public Result noEstimate(Metric metric)
+        public <T> Checks noEstimate(Metric<T> metric)
         {
-            return testMetrics(metric, metricComparison -> metricComparison.result(defaultTolerance()) == NO_ESTIMATE);
+            checks.add(new MetricsCheck<>(metric, (actual, estimate) -> true, NO_ESTIMATE));
+            return this;
+        }
+
+        public <T> Checks noBaseline(Metric<T> metric)
+        {
+            checks.add(new MetricsCheck<>(metric, (actual, estimate) -> true, NO_BASELINE));
+            return this;
         }
 
-        public Result noBaseline(Metric metric)
+        void run(@Language("SQL") String query, DistributedQueryRunner runner)
         {
-            return testMetrics(metric, metricComparison -> metricComparison.result(defaultTolerance()) == NO_BASELINE);
+            Set<Metric<?>> metrics = checks.stream()
+                    .map(check -> check.metric)
+                    .collect(toImmutableSet());
+            Set<MetricComparison<?>> metricComparisons = getMetricComparisons(query, runner, metrics);
+            for (MetricsCheck<?> check : checks) {
+                testMetrics(check.metric, metricComparison -> metricComparison.result(check.strategy) == check.expectedComparisonResult, metricComparisons);
+            }
         }
 
-        private Result testMetrics(Metric metric, Predicate<MetricComparison> assertCondition)
+        private Checks testMetrics(Metric metric, Predicate<MetricComparison> assertCondition, Set<MetricComparison<?>> metricComparisons)
         {
-            MetricComparison metricComparison = outputNodeMetricComparisons.get(metric);
-            assertTrue(assertCondition.test(metricComparison), "Following metrics do not match: " + metricComparison);
+            List<MetricComparison> testMetrics = metricComparisons.stream()
+                    .filter(metricComparison -> metricComparison.getMetric().equals(metric))
+                    .collect(toImmutableList());
+            assertTrue(testMetrics.size() > 0, "No metric found for: " + metric);
+            assertTrue(testMetrics.stream().allMatch(assertCondition), "Following metrics do not match: " + testMetrics);
```

Comment:
message is wrong. It suggests that these metrics are wrong, while the truth is that only some (in corners case all) of them are wrong.

It would be nice to print only wrong ones. But it could be done someday in the future.